### PR TITLE
orcidsync: count tokens to be synch'ed

### PR DESCRIPTION
**Pull request by puntonim**
Count the number of ORCID tokens (and relative data) to be synch'ed (and thus sent to Redis). This is used at Lab's side to check weather the right number of token has been processed.

### CODE REVIEW

File: bibtasklets/bst_orcidsync.py [hunk 1]
```diff
@@ -49,6 +64,25 @@ def orcid_data_to_sync():
             yield orcid.decode('utf8'), token.decode('utf8'), user_info['email'].decode('utf8'), name
 
 
+def count_orcid_data_to_sync():
+    """
+    Count the number of ORCID tokens (and relative data) to be synch'ed (and
+    thus sent to Redis). This is used at Lab's side to check weather the right
```
Comment (tsgit) [inline bibtasklets/bst_orcidsync.py]: typo: whether

File: bibtasklets/bst_orcidsync.py [hunk 2]
```diff
@@ -49,6 +64,25 @@ def orcid_data_to_sync():
...
+    number of token has been processed.
+
+    Returns (int): a number.
+    """
+    query = """
+        SELECT count(*) FROM (
+            SELECT personid, count(*) AS cnt FROM inspirehep.aidPERSONIDDATA WHERE tag='uid' AND personid IN (
```
Comment (tsgit) [inline bibtasklets/bst_orcidsync.py]: using a hardcoded `databasename` is not advised. while I understand the motivation to fully specify db+table the db name is configurable in `invenio.conf`. On my test install it's `inspiredemo` for example, or `inspiredev`. you get the idea. `run_sql` reads the proper db name from config, so you shouldn't specify it at all for portability.

File: bibtasklets/bst_orcidsync.py [hunk 3]
```diff
@@ -49,6 +64,25 @@ def orcid_data_to_sync():
...
+                SELECT personid FROM inspirehep.aidPERSONIDDATA NATURAL JOIN inspirehep.aidTOKEN WHERE tag='extid:ORCID'
+            ) GROUP BY personid
+        ) AS Q1;    
+    """
+    count = run_sql(query)
+    return int(count[0][0])
```
Comment (tsgit) [inline bibtasklets/bst_orcidsync.py]: that's somewhat fragile if the db query failed (intermittent outage, etc. yesterday I got an *out of memory* error on a query against `aidPERSONIDPAPERS`), but then it's probably wise the abort with exception anyhow. so I won't insist on try/except around it.

Comment (tsgit) [reply bibtasklets/bst_orcidsync.py]: here's a question: What ORCID profile is synced to when there are multiple ORCIDs for the same personid? This is possible in ORCID. currently there is only one such personid in Inspire:
```
|   419277 | extid:ORCID     | 0000-0001-8653-3761 | NULL | NULL | NULL | 2016-08-24 15:41:01 | NULL                                                                                                                 |
|   419277 | extid:ORCID     | 0000-0001-9267-1693 | NULL | NULL | NULL | 2017-10-04 02:10:21 | NULL                                                                                                                 |
|   419277 | uid             | 3877                |    0 |    0 |      | 0000-00-00 00:00:00 | NULL                                                                                                                 |```

Comment (tsgit) [reply bibtasklets/bst_orcidsync.py]: meanwhile Heath cleaned up that profile, so there's only one orcid left. they appeared to belong to completely different names anyhow.  so the question is moot.

File: bibtasklets/bst_orcidsync.py (hunk 1 above)
Comment (puntonim) [reply bibtasklets/bst_orcidsync.py]: Fixed

File: bibtasklets/bst_orcidsync.py (hunk 2 above)
Comment (puntonim) [reply bibtasklets/bst_orcidsync.py]: Fixed (I forgot to remove the db name when I copy/pasted from my client)

File: bibtasklets/bst_orcidsync.py (hunk 3 above)
Comment (puntonim) [reply bibtasklets/bst_orcidsync.py]: Legacy should not allow multiple ORCID by design. So the source of the problem is there.
Anyhow: legacy would push both ORCID profiles to Redis. Labs would read them, but store only the first ORCID per email address and then push with that ORCID.

**Pull request merged**